src/openms/source/ANALYSIS/ID/PercolatorFeatureSetHelper.cpp

```diff
@@ -287,6 +287,142 @@ namespace OpenMS
+                if (is_e_value) score = -log(score + 0.0001);
+
+                // different modification
+                if (ph->getSequence().toString() != sequence.toString())
+                {
+                  if (!top_mod_hit_found)
+                  {
+                    sibling_modifications += score;
+                    top_mod_hit_found = true;
+                  }
+                }
+
+                // different charge
+                if (ph->getCharge() != hit->getCharge())
+                {
+                  if (!top_ion_hit_found)
+                  {
+                    sibling_ions += score;
+                    top_ion_hit_found = true;
+                  }
+                }
+              }
+            }
+          }
+          hit->setMetaValue("REP:siblingSearches", sibling_searches);
+          hit->setMetaValue("REP:siblingModifications", sibling_modifications);
+          hit->setMetaValue("REP:siblingIons", sibling_ions);
+        }
+      }
+
+      int outer_index = 0;
+      for (vector<PeptideIdentification>::iterator outer = peptide_ids.begin(); outer != peptide_ids.end(); ++outer)
+      {
+        outer_index += 1;
+        double mz_tol = outer->getMZ() * 1e-5;  // 10ppm tolerance
+
+        for (vector<PeptideHit>::iterator outer_pep = outer->getHits().begin(); outer_pep != outer->getHits().end(); ++outer_pep)
+        {
+          double replicate_spectra = 0;
+          double sibling_experiments = 0;
+
+          int inner_index = 0;
+          for (vector<PeptideIdentification>::iterator inner = peptide_ids.begin(); inner != peptide_ids.end(); ++inner)
+          {
+            inner_index += 1;
+
+            bool is_e_value = inner->getScoreType() == "SpecEValue" || inner->getScoreType() == "expect" || inner->getScoreType() == "EValue";
+            bool mz_in_range = (inner->getMZ() >= outer->getMZ() - mz_tol) && (inner->getMZ() <= outer->getMZ() + mz_tol);
+            bool rt_in_range = (inner->getRT() >= outer->getRT() - 1200) && (inner->getRT() <= outer->getRT() + 1200);
+
+            // different identifications with similar m/z and RT
+            if (outer_index != inner_index && mz_in_range && rt_in_range)
+            {
+              for (vector<PeptideHit>::iterator inner_pep = inner->getHits().begin(); inner_pep != inner->getHits().end(); ++inner_pep)
+              {
+
+                // two peptide hits with identical charge and sequence
+                if (outer_pep->getCharge() == inner_pep->getCharge() &&
+                    outer_pep->getSequence().toUnmodifiedString() == inner_pep->getSequence().toUnmodifiedString())
+                {
+                  double score = inner_pep->getScore();
+                  if (is_e_value) score = -log(score + 0.0001);
+
+                  if (outer_pep->getMetaValue("TMP:sources") == inner_pep->getMetaValue("TMP:sources"))  // PSMs from different runs
+                  {
+                    replicate_spectra += score;
```

Comment:
did you also test absolute counts instead of score?
is the comparison correct? if the source is the same, shouldn't it be the same run?

Reply:
The code is correct - the comments should be swapped.

For features taking into account the precursor ion, I did not test absolute counts. Do you think that would work better?

Reply:
hmm I have no idea. probably not but if it is not much work may be worth a shot.

Reply:
Ok. Can you check and correct the comments in the code? I can update the pyOpenMS part if you want. In these dataset it indeed seems to be the case that these features are not very important. How costly are these to calculate?

Reply:
Sorry - I was busy writing my thesis. I'm going to correct the comments, update the pyOpenMS part, and implement the new features asap.

I'd say it takes like 5-10 minutes - so, it's not that bad (but a bit annoying).